ros_ws/src/teleoperation/src/joystick_controller.cpp

```diff
@@ -14,8 +17,21 @@ JoystickController::JoystickController()
     this->m_joystick_subscriber =
         this->m_node_handle.subscribe<sensor_msgs::Joy>("joy", 10, &JoystickController::joystickCallback, this);
 
+
+    // get the provided gamepad type
     ros::NodeHandle private_node_handle("~");
-    private_node_handle.getParam(INVERT_STEERING_PARAMETER, this->m_invert_steering);
+    private_node_handle.getParam(GAMEPAD_TYPE_PARAMETER, this->m_gamepad_type);
+
+    // load joystick map for the provided gamepad type
+    ROS_ASSERT_MSG(m_gamepad_type != "xbox360" || m_gamepad_type != "ps3", "Gamepad type should be xbox360 or ps3");
+    if (m_gamepad_type == "xbox360")
+    {
+        m_joystick_map = new JoystickMapXbox360();
```

Comment:
With the `new` you create a memory leak. Either add a `delete m_joystick_map;` to the destructor or use ownership-semantics by changing the type of  `m_joystick_map` to `std::unique_ptr<JoystickController>` and replace the `new JoystickMapXbox360();` with `std::make_unique<JoystickMapXbox360>();`